### PR TITLE
Shapes for Gantt, Deck and Portal

### PR DESCRIPTION
**Pull request by lfasani**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [x] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/adrs/124_support_deck_representation.adoc [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+* Daily
+* OKR
+
+== Decision
+
+For the front-end, the deck representation will rely on the https://github.com/rcdexta/react-trello/[react-trello project].
+
+We will create dedicated concepts for describing a Deck representation. It is called `Deck Description`. `Deck Description` concepts will be part of the `View` model.
+
+We will provide three instances/example of `Deck Description` corresponding to Kanban, Daily et OKR. These representation description will represent the `Task` meta-model.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: These representation description =>  These representation descriptions

File: doc/adrs/124_support_deck_representation.adoc [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+=== Overview of a Deck representation
+
+A Deck representation displays vertical lanes in which there are cards.
+
+image::./images/124/deck_overview.png[]
+
+=== General actions related to representation
+
+Like others representations in Sirius web, the Deck representation can be created from contextual menu on semantic object displayed in the Explorer tree.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: Like other (no S)

Comment (florianbarbin) [reply doc/adrs/124_support_deck_representation.adoc]: can be created from **a** contextual menu

Comment (florianbarbin) [reply doc/adrs/124_support_deck_representation.adoc]: semantic object**s** displayed

File: doc/adrs/124_support_deck_representation.adoc [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+* *Zoom +/-* : to zoom in and out
+* *Show/Hide* Lane (with drop down menu) : it allows to choose which columns are displayed. This information is stored in the representation.
+* *Share* : it allow to copy an URL in the clipboard corresponding to the diagram
+
+
+=== Actions associated to Lanes
+
+image:images/124/deck_lane_actions.png[Lane actions,600]
+
+* *Selection*: When clicking on the `Lane` if a semantic element is associated to it, its properties are dispayed in `Details` and the semantic element is selected in the `Explorer`.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: dispayed => displayed

File: doc/adrs/124_support_deck_representation.adoc [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+* *Add a card*: If configured in the view model, it is possible to add a Card.
+* *Hide Lane*: If configured in the view model, it is possible to hide a Lane. This information is stored in the representation.
+* *Visible Cards*: If configured in the view model, it is possible to choose which cards are displayed in the Lane. This information is stored in the representation.
+* *Delete*: If configured in the view model, it is possible to delete a Lane.
+
+
+
+=== Actions associated to Cards
+
+* *Selection*: When clicking on the `Card` the properties of the associated semantic element are dispayed in `Details` and the semantic element is selected in the `Explorer`.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: dispayed => displayed

File: doc/adrs/124_support_deck_representation.adoc [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+* enable edition
+* enable hide/show
+* enable deletion
+* drag and drop
+
+==== Understanding the way the representation is rendered
+
+Some tools will do modification in the representation itself such as Hide/Show Lane/Card, Expand/Collapse a Lane.
+
+For the other tools, the change is never done directly on the representation iteself. Instead, the change is done on the semantic model through the operations associated to the tool defined in the view model and then the representation will update based on the current state of the semantic model.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: iteself > itself

File: doc/adrs/124_support_deck_representation.adoc [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+Some tools will do modification in the representation itself such as Hide/Show Lane/Card, Expand/Collapse a Lane.
+
+For the other tools, the change is never done directly on the representation iteself. Instead, the change is done on the semantic model through the operations associated to the tool defined in the view model and then the representation will update based on the current state of the semantic model. 
+For example, when the user drag&drop a card from a lane to another, the expression should update the semantic model in a manner that at the end of the rendering the card get displayed in the lane where it has been dropped.
+
+=== Concepts of the Task meta-model
+
+The deck representation is agnostic from any meta-model. A deck representation can represent any semantic concept.
+Neverthless, in the idea of providing a ready to use fonctionality, we will provide a meta-model that matches the concept Deck.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: Nevertheless

Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: fonctionality => functionality

File: doc/adrs/124_support_deck_representation.adoc [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+This the Task meta-model.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: This is the

File: doc/adrs/124_support_deck_representation.adoc [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+* enable hide/show
+* enable deletion
+* drag and drop
+
+==== Understanding the way the representation is rendered
+
+Some tools will do modification in the representation itself such as Hide/Show Lane/Card, Expand/Collapse a Lane.
+
+For the other tools, the change is never done directly on the representation iteself. Instead, the change is done on the semantic model through the operations associated to the tool defined in the view model and then the representation will update based on the current state of the semantic model. 
+For example, when the user drag&drop a card from a lane to another, the expression should update the semantic model in a manner that at the end of the rendering the card get displayed in the lane where it has been dropped.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: the rendering the card get**s** displayed

File: doc/adrs/124_support_deck_representation.adoc [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+Neverthless, in the idea of providing a ready to use fonctionality, we will provide a meta-model that matches the concept Deck.
+This the Task meta-model.
+It is also described in the "Support Gantt representation" adr
+
+Let's focus here on the Task concept that will be used for Deck representation.
+
+* `Project` contains `Objective`s.
+* `AbstractTask` is specialized by `Objective`, `KeyResult` 
+*  `Objective` contains a list of `KeyResult` which contains a list of `Task` corresponding to initiatives.
+* `AbstractTask` have tags that will be used to sort tasks in different lanes for Daily and Kanban. For example, if the Task own the Tag "In work", the corresponding card will be displayed in the "In work" Lane in Kanban.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: **has** tags that will be used

Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: if the Task own**s** the

File: doc/adrs/124_support_deck_representation.adoc [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+image::images/124/taskMMForDeck.png[alt]
+
+=== Deck descriptions for Kanban, Daily and OKR
+
+We will provide three instance of `DeckDescription` in the view model. One for each representation.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: instances

File: doc/adrs/124_support_deck_representation.adoc [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+The provided `DeckDescription` instances will rely on the `Task` meta-model.
+
+==== Kanban
+ 
+ The Kanban representation is related to a `Project`.
+ Each Lane will represent a `TaskTag` which category is `kanban` and will contain as many cards which use this tag.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: which => whose

File: doc/adrs/124_support_deck_representation.adoc [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+
+==== Daily
+ 
+The Daily representation is related to a `Project`.
+Each Lane will represent a `TaskTag` which category is `daily` and will contain as many cards which use this tag.
```
Comment (florianbarbin) [inline doc/adrs/124_support_deck_representation.adoc]: which => whose

File: doc/adrs/124_support_deck_representation.adoc (hunk 1 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 2 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 3 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 4 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 5 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 6 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 7 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 8 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 9 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 11 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 12 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+We will create a dedicated Task meta-model.
+We will provide an instance/example of `Gantt Description` that will represent the `Task` model.
+The `Task` meta-model is a new meta-model that will also be provided.
+
+== Specification
+
+=== Overview of a Deck representation
+
+A Gantt representation displays tasks that are linked by dependencies.
+// TODO picture to update
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: TODO remaining

File: doc/adrs/123_support_gantt_representation.adoc [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+* rename the representation
+* delete the representation.
+
+=== Concepts of Gantt representation
+
+A gantt displays `Task`
+
+* There are three kinds of `Task`
+** simple task: a basic task which start and end date can be configured
+** group task: a task that can contains other tasks. The start and end dates are computed from the contained tasks or may be fixed.
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: For readable reason, we only keep one line per sentence in .adoc

File: doc/adrs/123_support_gantt_representation.adoc [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+A `Task` owns the following data:
+
+* Name
+* Description
+* Start date
+* End date
+* Progress
+* Assignee
+
+A task can depend on one or more another tasks. A dependency is always from the end of a task to the begining of the dependent task (no start-start, start-end, end-end).
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: One line per sentence

File: doc/adrs/123_support_gantt_representation.adoc [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+
+=== Display 
+
+The tasks are displayed in a grid which time unit is changeable. See Zoom level in next chapter.
+
+The tasks are displayed on one line as a rectangle containing the name of the Task. The name is displayed beside the task if it is too long.
+There is one color for the task itself and one color corresponding to the progress.
+The dependency relation is represented by an oriented arrow.
+
+The current date  is displayed with a special color(light yellow in the example)
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: double space between _date_ and _is_

Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: missing space after color

File: doc/adrs/123_support_gantt_representation.adoc [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+A task can depend on one or more another tasks. A dependency is always from the end of a task to the begining of the dependent task (no start-start, start-end, end-end).
+A task has a progress (0 to 100%)
+
+A task is associated to a semantic model element(visible in the Explorer tree).
+
+image::images/123/gantt_tasks.png[Gantt tasks]
+
+=== Display 
+
+The tasks are displayed in a grid which time unit is changeable. See Zoom level in next chapter.
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: One line per sentence

File: doc/adrs/123_support_gantt_representation.adoc [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+
+The tasks are displayed on one line as a rectangle containing the name of the Task. The name is displayed beside the task if it is too long.
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: One line per sentence

File: doc/adrs/123_support_gantt_representation.adoc [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+* Assignee
+
+=== Toolbar actions
+
+image::images/123/gantt_display_actions.png[Display actions,600]
+
+* Zoom level : it is possible to select the zoom level
+* Zoom +/- : to zoom in and out
+* Fit to screen : the zoom level is chosen so that all tasks can be seen on the screen
+* Columns : clicking on it will hide all the columns. Clicking on drop dow menu will allow to choose the displayed columns
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: One line per sentence

File: doc/adrs/123_support_gantt_representation.adoc [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+
+The Gantt representation is related to a `Project`. Each gantt task represent a `Task`. The gantt task attributes are the same as the `Task`.
+
+=== Limitations
+
+It would be interesting to have start and end handles when selecting a dependency. Then moving an handle from a task to another would allow to create or change the dependency.
+
+== Status
+
+In work
```
Comment (frouene) [inline doc/adrs/123_support_gantt_representation.adoc]: Can be changed to accepted ?

File: doc/adrs/124_support_deck_representation.adoc [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+
+* Kanban
+* Daily
+* OKR
+
+== Decision
+
+For the front-end, the deck representation will rely on the https://github.com/rcdexta/react-trello/[react-trello project].
+
+We will create dedicated concepts for describing a Deck representation. It is called `Deck Description`. `Deck Description` concepts will be part of the `View` model.
```
Comment (frouene) [inline doc/adrs/124_support_deck_representation.adoc]: One line per sentence

File: doc/adrs/124_support_deck_representation.adoc [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+The Daily representation is related to a `Project`.
+There are three Lanes:
+
+* One to represent the Objectives
+* One to represent the KeyResults
+* One to represent the Initiatives
+
+== Status
+
+In work
```
Comment (frouene) [inline doc/adrs/124_support_deck_representation.adoc]: Can be changed to accepted ?

File: doc/adrs/124_support_deck_representation.adoc [hunk 23]
```diff
@@ -0,0 +1,170 @@
...
+
+=== Deck descriptions for Kanban, Daily and OKR
+
+We will provide three instance of `DeckDescription`. One for each representation.
+The provided `DeckDescription` instances will rely on the `Task` meta-model.
+
+==== Kanban
+ 
+The Kanban representation is related to a `Project`.
+Each Lane will represent a `TaskTag` whose category is `kanban` and will contain as many cards that use this tag.
```
Comment (frouene) [inline doc/adrs/124_support_deck_representation.adoc]: Does the concept of category for tag need to be described somewhere?

File: doc/adrs/124_support_deck_representation.adoc [hunk 24]
```diff
@@ -0,0 +1,170 @@
...
+
+* title
+* label
+
+A card has
+
+* title
+* label
+* description
+* progress
```
Comment (frouene) [inline doc/adrs/124_support_deck_representation.adoc]: The progress is no display on the card ?

File: doc/adrs/123_support_gantt_representation.adoc (hunk 13 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 14 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 15 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 16 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 17 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 18 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 19 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: done

File: doc/adrs/123_support_gantt_representation.adoc (hunk 20 above)
Comment (lfasani) [reply doc/adrs/123_support_gantt_representation.adoc]: yes, done

File: doc/adrs/124_support_deck_representation.adoc (hunk 21 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done and everywhere else in the file

File: doc/adrs/124_support_deck_representation.adoc (hunk 23 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: good catch.
`TaskTag` is a concept of Task MM so I added a line to describe it in `=== Concepts of the Task meta-model` paragraph

File: doc/adrs/124_support_deck_representation.adoc (hunk 22 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: done

File: doc/adrs/124_support_deck_representation.adoc (hunk 24 above)
Comment (lfasani) [reply doc/adrs/124_support_deck_representation.adoc]: yes effectively I think progress should not be card concept.
Freely the specifier can display the progress in the label if progress is a semantic concept.
So I am removing progress concept from Card

File: doc/adrs/122_add_support_for_portal_representation.adoc [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+. share button (opens a popup with the URL to access to the _portal_ representation)
+. toggle to edit mode
+. toggle to direct mode
+
+=== Representations panel
+
+* The right panel of the sirius page includes a representation view.
+* For _portal_, this view presents, in the form of a tree, the representations it contains.
+* No changes to a _portal_ content can be made from this view.
+
```
Comment (lfasani) [inline doc/adrs/122_add_support_for_portal_representation.adoc]: The trash icon should be removed from the screen shot

Comment (frouene) [reply doc/adrs/122_add_support_for_portal_representation.adoc]: The last trash icons are for the main representation, it's the current behavior.
I only remove the one for the sub-representation

**Review by pcdavid — CHANGES_REQUESTED** [review]
These should be shapes, not ADRs.

Shapes are for functional specification, which is the case here.

ADRs should be reserved to discussion and justification of technical implementation choices and their impact (originally they are even supposed to be reserved for *architectural* decisions, but that not how we use them in practice).

**Comment by frouene** [comment]
I've changed this documentation for shapes

**Review by pcdavid — APPROVED** [review]

**Pull request merged**